TestFoundation/TestOperationQueue.swift

```diff
@@ -168,34 +168,42 @@ class TestOperationQueue : XCTestCase {
 class AsyncOperation: Operation {
 
     private let queue = DispatchQueue(label: "async.operation.queue")
-    private let lock = NSLock()
+    private let lock = NSRecursiveLock()
```

Comment:
For KVO in objc, the last wins and the will change is practically not used. This is of course a non-issue since the result should only transition to a valid state and any invalid transition e.g. from finished back to not finished are non-sense in the first place.

The objective-c implementation of NSOperationQueue assumes that the state machine transition of operations is always forward progressing. Any backwards movement is considered either a) invalid by nature and will halt or b) ignorable since it is nonsense.